applications/ShapeOptimizationApplication/custom_utilities/vertex_morphing_mapper.h

```diff
@@ -168,87 +165,90 @@ class VertexMorphingMapper
+                numberOfNodesAffected = treeOfAllOptimizationNodes.SearchInRadius( currentDampingNode, 
+                                                                                   dampingRadius, 
+                                                                                   nodesAffected.begin(),
+                                                                                   resulting_squared_distances.begin(), 
+                                                                                   maxNodesAffected ); 
                 
                 // Throw a warning if specified (hard-coded) maximum number of neighbors is reached
-                if(number_of_nodes_affected == max_nodes_affected)
-                    std::cout << "\n> WARNING!!!!! For node " << damping_node_i.Id() << " and specified damping radius, maximum number of neighbor nodes (=" << max_nodes_affected << " nodes) reached!" << std::endl;
+                if(numberOfNodesAffected == maxNodesAffected)
+                    std::cout << "\n> WARNING!!!!! For node " << currentDampingNode.Id() << " and specified damping radius, maximum number of neighbor nodes (=" << maxNodesAffected << " nodes) reached!" << std::endl;
 
                 // Loop over all nodes in radius (including node on damping region itself)
-                for(unsigned int j_itr = 0 ; j_itr<number_of_nodes_affected ; j_itr++)
+                for(unsigned int j_itr = 0 ; j_itr<numberOfNodesAffected ; j_itr++)
                 {
                     // Get node information
-                    ModelPart::NodeType& node_j = *nodes_affected[j_itr];
+                    ModelPart::NodeType& node_j = *nodesAffected[j_itr];
                     array_3d j_coord = node_j.Coordinates();
 
                     // Computation of damping factor
-                    double damping_factor = damping_function.compute_damping_factor(i_coord,j_coord);
+                    double dampingFactor = dampingFunction.compute_damping_factor(i_coord,j_coord);
 
                     // For every specified damping direction we check if new damping factor is smaller than the assigned one for current node. 
                     // In case yes, we overwrite the value. This ensures that the damping factor of a node is computed by its closest distance to the damping region
-                    if(damp_in_X == true && damping_factor < node_j.GetValue(DAMPING_FACTOR_X))
-                        node_j.SetValue(DAMPING_FACTOR_X, damping_factor);     
-                    if(damp_in_Y == true && damping_factor < node_j.GetValue(DAMPING_FACTOR_Y))       
-                        node_j.SetValue(DAMPING_FACTOR_Y, damping_factor);   
-                    if(damp_in_Z == true && damping_factor < node_j.GetValue(DAMPING_FACTOR_Z))       
-                        node_j.SetValue(DAMPING_FACTOR_Z, damping_factor);                            
+                    if(dampX == true && dampingFactor < node_j.GetValue(DAMPING_FACTOR_X))
```

Comment:
i would suggest to get 
    auto& v = node.GetValue(DAMPING_FACTOR)

and then to change the components as
v[0] = ...
v[1] = ...

this is much faster since you access to the database just once